### PR TITLE
JDBC Sources: remove redundant date-time datatypes formatting

### LINKED ISSUE #21801: Date Time Parsing Slowness in Postgres Source
Based on the Postgres profiling report, we realised that parsing timestamp values in Postgres db source is a costly operation. 
It is contributing to a lot of allocation and CPU time as well. 

[profiling_report.zip](https://github.com/airbytehq/airbyte/files/10492329/profiling_report.zip)

Please note that the above profiling was done with a version of Postgres that contained changes from PRs
https://github.com/airbytehq/airbyte/pull/21691
https://github.com/airbytehq/airbyte/pull/21634
https://github.com/airbytehq/airbyte/pull/21523

### PR DESCRIPTION
**Pull request by VitaliiMaltsev**
## What
JDBC sources use DateTimeFormatter to format value fetched from the db
In most cases it's redundant due to ResultSet already produced ISO8601 value

## How
*Describe the solution*

## Recommended reading order
1. `AbstractJdbcCompatibleSourceOperations.java`
2. `DateTimeConverter.java`

## 🚨 User Impact 🚨
Are there any breaking changes? What is the end result perceived by the user? If yes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [x] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [x] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

<details><summary><strong>Connector Generator</strong></summary>

- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed

</details>

## Tests

<details><summary><strong>Unit</strong></summary>

*Put your unit tests output here.*

</details>

<details><summary><strong>Integration</strong></summary>

*Put your integration tests output here.*

</details>

<details><summary><strong>Acceptance</strong></summary>

*Put your acceptance tests output here.*

</details>

### CODE REVIEW

**Comment by akashkulk** [comment]
Are there cases where the `ResultSet` associated with JDBC sources **_don't_** produce the ISO8601 value? Want to make sure we understand/document and don't break those cases

**Comment by VitaliiMaltsev** [comment]
> Are there cases where the `ResultSet` associated with JDBC sources **_don't_** produce the ISO8601 value? Want to make sure we understand/document and don't break those cases

@akashkulk ResultSet can produce non ISO8601 value in case of SQL formatting function is used
For example

`Select to_char(some_date_column,'MM/YY/YYYY') from test_date`

but in airbyte this is not used anywhere, so we always get the default value of the JDBC driver (ISO8601)

File: airbyte-db/db-lib/src/main/java/io/airbyte/db/jdbc/DateTimeConverter.java [hunk 1]
```diff
@@ -98,24 +98,24 @@ public static String convertToTimestamp(final Object timestamp) {
...
       if (!loggedUnknownTimestampClass) {
         LOGGER.info("Unknown class for Timestamp data type" + timestamp.getClass());
         loggedUnknownTimestampClass = true;
       }
       final LocalDateTime localDateTime = LocalDateTime.parse(timestamp.toString());
       final LocalDate date = localDateTime.toLocalDate();
-      final String value = localDateTime.format(TIMESTAMP_FORMATTER);
-      return resolveEra(date, value);
+      return resolveEra(date,
+          hasZeroSecondsAndNanos(localDateTime.toLocalTime()) ? localDateTime.format(TIMESTAMP_FORMATTER) : localDateTime.toString());
```
Comment (rodireich) [inline airbyte-db/db-lib/src/main/java/io/airbyte/db/jdbc/DateTimeConverter.java:118]: nit: If it makes sense, use formatTime()

**Review by rodireich — APPROVED** [review]

**Comment by VitaliiMaltsev** [comment]
/publish connector=connectors/source-clickhouse-strict-encrypt

**Comment by VitaliiMaltsev** [comment]
/publish connector=connectors/source-mssql-strict-encrypt run-tests=false

**Pull request merged**